Make unit-tests run with `woocommerce/date` (~`/components`~) dependency.

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Transform `@woocommerce/data` build for jest,
	  To make our tests run with `@woocommerce/date` ~or `@woocommerce/components`~ dependencies.
	  Fixes **part of** https://github.com/woocommerce/google-listings-and-ads/pull/234.
	  Works around **part of**  https://github.com/woocommerce/woocommerce-admin/issues/6483.
- Removes no longer needed `woocommerce/date` mock.


### Screenshots:

![Screenshot from test run results](https://user-images.githubusercontent.com/17435/118597605-3026b380-b7ad-11eb-9829-0366582afd06.png)


### Detailed test instructions:
```
git co fix/234-wc-admin-tests
npm install
npm run test-unit
```

Make sure tests run without complaining about `@woocommerce/date` dependency, like:
```
 SyntaxError: Cannot use import statement outside a module

      2 |  * External dependencies
      3 |  */
    > 4 | import { getCurrentDates } from '@woocommerce/date';
```

### Changelog Note:

> Workaround `@woocommerce/date` breaking tests.


### Additional notes:

Shall we remove the incomplete e2e test and enable `npm test-unit` to run in travis?

tomalec committed:
Transform `@woocommerce/data` build for jest,

To make our tests run with `@woocommerce/date` or `@woocommerce/components` dependencies.

Fixes https://github.com/woocommerce/google-listings-and-ads/pull/234.
Works around https://github.com/woocommerce/woocommerce-admin/issues/6483.

eason9487 reviewed (commented):
❓ 
It still has the same problem as #234. Seems some dependencies under `@woocommerce`, e.g. `@woocommerce/components`, are not addressed by this PR.

Add `import { Spinner } from '@woocommerce/components';` to the [hello/index.js](https://github.com/woocommerce/google-listings-and-ads/blob/39f9592696375f4f6ee5d0d166e80d5638a92970/js/src/hello/index.js) and run `npm run test-unit`:

![image](https://user-images.githubusercontent.com/17420811/118639774-d7860380-b80a-11eb-87ad-16f2fa5ffcd7.png)

tomalec committed:
Remove not needed transform ifrom jest.config

Addresses: https://github.com/woocommerce/google-listings-and-ads/pull/622#discussion_r634264178

eason9487 reviewed (approved):
LGTM.
